examples/spring-boot-with-certmanager-example/README.md

```diff
@@ -0,0 +1,122 @@
+# Spring Boot with Cert-Manager
+
+The purpose of this example is to demonstrate how you can use the Cert-Manager Dekorate extension with a Spring Boot application.
+
+Apart from the Kubernetes Dekorate Starter dependency, this application is also using the Cert-Manager dekorate dependency:
+
+    <dependency>
+      <groupId>io.dekorate</groupId>
+      <artifactId>certmanager-annotations</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+
+## Enable HTTPS transport
+
+To enable the HTTPS transport in Spring Boot, we need to add the following properties:
+
+```
+server.port=8443
+server.ssl.enabled=true
+```
+
+Next, we need to configure the Keystore that our Spring Boot application will use to validate the user requests. In this example, we'll use a PKCS12 Keystore type:
+
+```
+server.ssl.key-store-type=PKCS12
+server.ssl.key-store=/path/to/keystore.p12
+server.ssl.key-store-password=the password
+```
+
+Where is the `keystore.p12` Keystore and what is its password? This is where Cert-Manager comes to play. Cert-Manager will generate a valid Keystore for your application to use.
+
+## Generate the Keystore
+
+In this section, we're going to generate the `keystore.p12` using the password `supersecret`.
+
+Let's start with the password first. We need to use a Secret resource with the value `supersecret` in the data key `password`:
+
+```
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: pkcs12-pass
+data:
+  # "supersecret" in base64:
+  password: c3VwZXJzZWNyZXQ=
+type: Opaque
+```
+
+In Dekorate, we can include these resources in the file `src/main/resources/k8s/common.yml`. To include these resources as part of the final kubernetes.yml file, we need to configure Dekorate as follows:
+
+```
+dekorate.options.input-path=k8s
+```
+
+Let's now configure the certificate to generate the previous Keystore:
+
+```
+dekorate.certificate.secret-name=tls-secret
+dekorate.certificate.self-signed.enabled=true
+dekorate.certificate.keystores.pkcs12.create=true
+# the secret name of the password (see above):
+dekorate.certificate.keystores.pkcs12.passwordSecretRef.name=pkcs12-pass 
+dekorate.certificate.keystores.pkcs12.passwordSecretRef.key=password
+```
+
+Using this configuration, Cert-Manager will generate a self-signed certificate and the Keystore files within the secret `tls-secret`. Therefore, we need to map the data within this secret `tls-secret` into the Spring Boot properties `server.ssl.key-store` and `server.ssl.key-store-password`.
+
+Moreover, the Cert-Manager Dekorate extension will configure the application resources to mount a volume with the generated certificates at the path `/etc/certs` (it can be configured using `dekorate.certificate.volume-mount-path`).
```

Comment:
... mount a volume with the generated certificates ... --->
... mount a volume with the generated PKCS12 keystore file ...